c10/util/BFloat16-inl.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <cstring>
+#include <limits>
+#include <c10/macros/Macros.h>
+
+namespace c10 {
+
+/// Constructors
+
+inline C10_HOST_DEVICE BFloat16::BFloat16(float value) {
+  val_ = detail::bf16_from_f32(value);
+}
+
+/// Implicit conversions
+
+inline C10_HOST_DEVICE BFloat16::operator float() const {
+  return detail::f32_from_bf16(val_);
+}
+
+// CUDA intrinsics
+
+#if defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 350)
+inline __device__ BFloat16 __ldg(const BFloat16* ptr) {
+    return __ldg(reinterpret_cast<const BFloat16*>(ptr));
```

Comment:
is this used?  This doesn't look correct.

Reply:
Weird. I was getting a lot of errors due to _ldg and explicit constructors but cant repro those anymore. Removing it until needed.

Reply:
It will be needed for [this](https://github.com/pytorch/pytorch/blob/master/aten/src/THC/THCTensorCopy.cu#L12).

Reply:
discussed offline: we dont need it and will follow the same path as bool does for the __ldg